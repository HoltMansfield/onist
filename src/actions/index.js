import * as types from '../constants/actionTypes'

export const addChild = child => ({ type: types.ADD_CHILD, child })
export const deleteChild = child => ({ type: types.DELETE_CHILD, child })
export const editChild = child => ({ type: types.EDIT_CHILD, child })

export const setQuestionState = newState => ({ type: types.SET_QUESTION_STATE, newState })
