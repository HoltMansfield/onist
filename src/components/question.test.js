import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme';
import reducer from '../reducers'
import Question from './question'
import { addChild } from '../actions'

const store = createStore(reducer)
const getComponentUnderTest = () => {
  return (
      <Provider store={store}>
        <Question />
      </Provider>
    )
}

test('question component rendering', () => {
  const component = renderer.create(getComponentUnderTest(),)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('question component caption with no children', () => {
  const wrapper = mount(getComponentUnderTest())
  const caption = wrapper.find(Question).find('div').find('span').first()

  expect(caption.text()).toEqual('Do you have any children?')
})

test('question component caption with 1 or more children', () => {
  const wrapper = mount(getComponentUnderTest())
  store.dispatch(addChild({}))
  const caption = wrapper.find(Question).find('div').find('span').first()

  expect(caption.text()).toEqual('Do you have any other children?')
})

// test('question component dispatches addChild action', () => {
//   const question = mount(getComponentUnderTest()).find(Question)
//
//   const spy = jest.spyOn(question.props, 'dispatch')
//   const yesButton = question.find('div').find('input').first()
//   yesButton.simulate('click')
//   expect(spy).toHaveBeenCalled()
// })
// test('question component dispatches addChild action', () => {
//   const question = mount(getComponentUnderTest()).find(Question)
//   expect(question.length).toEqual(1)
//   const spy = jest.spyOn(question.instance(), 'handleYesClicked')
//   const yesButton = question.find('div').find('input').first()
//   yesButton.simulate('click')
//   expect(spy).toHaveBeenCalled()
// })

test('question component dispatches addChild action', () => {
  const wrapper = mount(getComponentUnderTest())
  const question = wrapper.find(Question)
  expect(question.length).toEqual(1)
  const spy = jest.spyOn(question.instance(), 'handleYesClicked')
  wrapper.update()
  const yesButton = question.find('div').find('input').first()
  yesButton.simulate('click')
  expect(spy).toHaveBeenCalled()
})
