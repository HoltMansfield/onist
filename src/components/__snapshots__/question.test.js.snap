// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question component rendering 1`] = `
<div>
  <span>
    Do you have any children?
  </span>
  <input
    className="caption"
    name="yesNo"
    onClick={[Function]}
    type="radio"
    value="1"
  />
  <span>
    Yes
  </span>
  <input
    name="yesNo"
    type="radio"
    value="0"
  />
  <span>
    No
  </span>
</div>
`;
